Transactions Page

Tyler-Yue opened the pull request:
References #7

## Draft PR

- added transactions table 
- currently the API endpoint used is not merged into prod so you need to run api_app.py from the Fund Monitor on localhost on branch: 66-get-transactions in order to see the data
- the CSV export is not currently working

CameronBeneteau reviewed (changes requested):
@Tyler-Yue Can you also edit the header and main page so transactions no longer shows as disabled?

CameronBeneteau reviewed (changes requested):
Looks great so far, can you also make sure the URL changes on initial load with the API parameters similar to holdings and performance? On my end it doesn't auto change to `transactions?portfolio=core` on initial page load, only after portfolio selection.

CameronBeneteau commented:
Assuming it's fine, but adding @rajpandya737 just in case he wants to verify auth.

rajpandya737 commented on src/app/transactions/page.tsx:
```diff
@@ -0,0 +1,267 @@
+
+export default function TransactionsPage() {
+    return (
+        <Suspense fallback={<div>Loading...</div>}>
```
Change the fallback here to be the loading component like it is in the other files so it doesn't load a black screen on startup

rajpandya737 reviewed (changes requested):
in the transactions page, make the fallback the loading spinner to avoid the black screen on startup

OlegGlo reviewed (changes requested):
Nice. 4 Things:
- Change date to YYYY-MM-DD format.
- Remove the portfolio column.
- Swap ticker column and name column.
- Add sorting when clicking the column headers similar to how Tvesha did it with the holdings page.

OlegGlo commented:
Also replace route to hit our backend now.

Tyler-Yue commented:
## Changes 

- [X] Added router from main page and header bar
- [X] Added automatic URL params on page load
- [X] Changed fallback to be Loading component
- [X] Changed date format to YYYY-MM-DD
- [X] Removed portfolio column
- [X] Swapped ticker and name column
- [X] Added sort functionality to columns
- [X] Changed routing to use backend

CameronBeneteau commented on src/app/transactions/page.tsx:
```diff
@@ -0,0 +1,332 @@
+                                                    row.action === 'SELL' ? theme.palette.error.main : 'inherit'
+                                            }}>{row.action}</TableCell>
+                                            <TableCell align="center" sx={{ fontSize: '1rem' }}>{row.shares.toLocaleString()}</TableCell>
+                                            <TableCell align="center" sx={{ fontSize: '1rem' }}>{Number(row.price).toFixed(2)}</TableCell>
```
Add dollar sign here (same as holdings where there's a space between)

OlegGlo commented on src/app/components/nav.tsx:
Final change, make buttons wide similar to current look. Currently too narrow

<img width="280" alt="image" src="https://github.com/user-attachments/assets/79308f61-5937-4823-95e9-5d77eb26dcc8" />

CameronBeneteau replied:
@Tyler-Yue @OlegGlo What about flex-row the buttons instead of flex-col? Looks pretty good imo

OlegGlo replied:
columns is good cause it maintains hierarchy of page importance

Tyler-Yue replied:
![Screenshot 2025-03-31 224005](https://github.com/user-attachments/assets/06cc03d0-aeb5-4d8f-b0cb-f5d6526bc021)

Width 250px, does this look good?

CameronBeneteau reviewed (changes requested):
Looks like default columns are the wrong size on reload while the data is loading. Is there any way to update them so they don't snap wider once data is loaded? It doesn't look like it happens for the performance page table right now. If too hard to fix not the end of the world, was just curious.

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/ca3cb4f7-38b2-4297-a7f6-f4dbcdf9db28" />

CameronBeneteau reviewed (approved):
Beauty 🤠

CameronBeneteau commented:
@Tyler-Yue Chaged `Security Name` column to `Name` to match other pages

OlegGlo commented:
Good stuff
